src/com/tenthbit/view/ZoomImageView.java

```diff
@@ -225,22 +225,25 @@ public boolean onTouchEvent(MotionEvent event) {
                         lastTouchY = y;
 
                         // Compute velocity within the last 1000ms
-                        velocityTracker.addMovement(event);
-                        velocityTracker.computeCurrentVelocity(1000);
+			if(velocityTracker != null) {
+                        	velocityTracker.addMovement(event);
+	                        velocityTracker.computeCurrentVelocity(1000);
 
-                        final float vX = velocityTracker.getXVelocity(), vY = velocityTracker
+        	                final float vX = velocityTracker.getXVelocity(), vY = velocityTracker
                                 .getYVelocity();
 
-                        // If the velocity is greater than minVelocity perform
-                        // a fling
-                        if ((Math.max(Math.abs(vX), Math.abs(vY)) >= scaledMinimumFlingVelocity)
-                                && (getDrawable() != null)) {
-                            currentFlingRunnable = new FlingRunnable(getContext());
-                            currentFlingRunnable.fling(getWidth(), getHeight(), (int) -vX,
-                                    (int) -vY);
-                            post(currentFlingRunnable);
-                        }
-                    }
+	                        // If the velocity is greater than minVelocity perform
+        	                // a fling
+                	        if ((Math.max(Math.abs(vX), Math.abs(vY)) >= scaledMinimumFlingVelocity)
+                        	        && (getDrawable() != null)) {
+                            		currentFlingRunnable = new FlingRunnable(getContext());
+	                            	currentFlingRunnable.fling(getWidth(), getHeight(), (int) -vX,
+        	                            (int) -vY);
+                	            	post(currentFlingRunnable);
+                        	} 
+		        }//if velocityTracker != null
```

Comment:
I would say remove these comments: //if velocityTracker != null and //if isDragging